process/block/preprocess/gasComputation.go

```diff
@@ -17,13 +17,17 @@ type gasComputation struct {
 	economicsFee  process.FeeHandler
 	txTypeHandler process.TxTypeHandler
 	//TODO: Refactor these mutexes and maps in separated structures that handle the locking and unlocking for each operation required
-	gasProvided            map[string]uint64
-	gasProvidedAsScheduled map[string]uint64
-	mutGasProvided         sync.RWMutex
-	gasRefunded            map[string]uint64
-	mutGasRefunded         sync.RWMutex
-	gasPenalized           map[string]uint64
-	mutGasPenalized        sync.RWMutex
+	gasProvided                                      map[string]uint64
```

Comment:
refactoring idea for next PRs in order to avoid having a lot of fields along with their protective mutexes (which, improperly used can cause deadlocks):
create ConcurrentSafeBytesSlice or ConcurrentSafeStringUint64Map that can internally hold the mutexes and have some exported functions that can safely lock/unlock the mutexes

Reply:
Yep, should be done this way